ip.go

```diff
@@ -85,13 +91,16 @@ const (
 // IPAddressReservation is created when user sends IP reservation request for a project (considering it's within quota).
 type IPAddressReservation struct {
 	IpAddressCommon
-	Assignments []*IPAddressAssignment `json:"assignments"`
-	Facility    *Facility              `json:"facility,omitempty"`
-	Available   string                 `json:"available"`
-	Addon       bool                   `json:"addon"`
-	Bill        bool                   `json:"bill"`
-	State       IPReservationState     `json:"state"`
-	Description *string                `json:"details"`
+	Assignments  []*IPAddressAssignment `json:"assignments"`
+	Facility     *Facility              `json:"facility,omitempty"`
+	Available    string                 `json:"available"`
+	Addon        bool                   `json:"addon"`
+	Bill         bool                   `json:"bill"`
+	State        IPReservationState     `json:"state"`
+	Description  *string                `json:"details"`
+	Enabled      bool                   `json:"enabled"`
+	MetalGateway *MetalGateway          `json:"metal_gateway,omitempty"`
```

Comment:
MetalGatewayLite is what is actually returned here.

This MetalGateway response is consistent with how we've added fields in the past, but we've been trying to do better about capturing when a Lite response is provided.

The Lite response, in this case omits a few of the properties present in MetalGateway.  IIRC, in other situations a Lite response can include fields not present in the more verbose type.

We don't have a MetalLite type defined today. Just thinking about this makes me want to embed MetalGatewayLite in MetalGateway.

I'll leave it to reviewers to demand a more substantial change here.

Reply:
Here's the difference provided by the generated SDK in gometal:

https://github.com/t0mk/gometal/blob/main/v1/model_metal_gateway.go#L19-L31 vs 
https://github.com/t0mk/gometal/blob/main/v1/model_metal_gateway_lite.go#L19-L31

Reply:
IMO I would leave MetalGateway which gives more flexibility and less dependency on little changes than the Lite object can have on the API. The `omitempty` tag will make the necessary filter with parameters not included in the Lite, and I don't know how it has been handled on previous situations, but it seems strange to me that extra fields are included in Lite, but if it was the case I will just include them in our parent struct, MetalGateway in this case

Reply:
I was web slinging URLs in my previous commit. I took another look and see that MetalGatewayLite does have some substantially different fields. The MetalGatewayLite response (which the IPReservation body is said to include) is missing fields found in MetalGateway (Project, CreatedBy) but also has different fields which I didn't notice at first glance:
- GatewayAddress (string) vs IPReservation (Href or Object)
- Vlan (VNID) vs VirtualNetwork (Href or Object)

It seems we'll need and should use MetalGatewayLite since it is not a subset of and otherwise does not align with MetalGateway fields.